.changeset/dirty-ears-push.md

```diff
@@ -0,0 +1,7 @@
+---
+'@hyperlane-xyz/infra': patch
+'@hyperlane-xyz/cli': patch
+'@hyperlane-xyz/sdk': patch
+---
+
+Added `shouldRecover` flag to deployContractFromFactory
```

Comment:
Nit: Adding a note about how this fixes testRecipient related errors would be a good inclusion here too